LibreNMS/Alert/Transport/Weechatbot.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/* This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>. */
+
+/**
+ * Weechat Bot Transport
+ *
+ * @author RobJE <epping@renf.us>
+ * @copyright 2025 RobJE
+ * @license GPL
+ */
+
+namespace LibreNMS\Alert\Transport;
+
+use LibreNMS\Alert\Transport;
+use LibreNMS\Exceptions\AlertTransportDeliveryException;
+
+class WeechatBot extends Transport
+{
+    protected string $name = 'Weechat Bot';
+
+    public function deliverAlert(array $alert_data): bool
+    {
+        $pre = "{$this->config['bot-password']} ";
+        if (isset($this->config['irc-server']) && strlen($this->config['irc-server']) !== 0) {
+            $pre .= "{$this->config['irc-server']} ";
+        }
+        $pre .= "{$this->config['irc-channel']} ";
+
+        // https://www.php.net/manual/en/function.fsockopen.php example #2
+        $fp = fsockopen('udp://' . $this->config['bot-hostname'], $this->config['bot-port'], $errno, $errstr);
```

Comment:
would you mind using https://github.com/clue/socket-raw ? we already have this package

Reply:
I don't mind. Lack the knowledge to say which one is best.

Looks like both are used. `fsockopen` is used 7 times. `socket-raw` is used 6 times of witch 4 counts is in tests.

Will try to change to `socket-raw`

Reply:
@Jellyfrog converted to `socket-raw` with the help of a coworker.

Still wondering what the benefits are in this case. Now there is a external dependency and a lot more code involved for basically the same functionality that `fsockopen` has. Its use is only within this one function as well.

The `fsockopen` approach more elegant to me.